### PR TITLE
[Feature-16801][UI] Add copy path of resource file in task definition

### LINKED ISSUE #16801: [Feature][UI] Add copy path of resource file in task definition
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Add copy path of resource file in task definition

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SbloodyS**
## Purpose of the pull request

close #16801

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

## Pull Request Notice
[Pull Request Notice](https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by caishunfeng — DISMISSED** [review]
LGTM

File: dolphinscheduler-storage-plugin/dolphinscheduler-storage-api/src/main/java/org/apache/dolphinscheduler/plugin/storage/api/StorageOperator.java [hunk 1]
```diff
@@ -24,9 +24,6 @@
 
 public interface StorageOperator {
 
-    String FILE_FOLDER_NAME = "resources";
-    String UDF_FOLDER_NAME = "udfs";
```
Comment (ruanwenjun) [inline dolphinscheduler-storage-plugin/dolphinscheduler-storage-api/src/main/java/org/apache/dolphinscheduler/plugin/storage/api/StorageOperator.java]: Please Keep these two fields, we should avoid add all constants in Constants, these fields in StorageOperator is also static. You can get these by `StorageOperator.FILE_FOLDER_NAME`

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/resources/visitor/ResourceTreeVisitor.java [hunk 2]
```diff
@@ -123,9 +124,15 @@ private static ResourceComponent getResourceComponent(StorageEntity resource) {
             tempResourceComponent = new FileLeaf();
         }
 
+        String currentDir = "";
+        if (resource.getFullName().contains(Constants.RESOURCE_TYPE_FILE)) {
+            currentDir = resource.getFullName().split(Constants.RESOURCE_TYPE_FILE)[1].substring(1);
+        }
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/resources/visitor/ResourceTreeVisitor.java]: This is not safe, we should parse the `ResourceMetaData` and check whether the `resource'resourceType` is `FILE`. You can find some code under `AbstractStorageOperator#getResourceMetaData`

File: dolphinscheduler-storage-plugin/dolphinscheduler-storage-api/src/main/java/org/apache/dolphinscheduler/plugin/storage/api/StorageOperator.java (hunk 1 above)
Comment (SbloodyS) [reply dolphinscheduler-storage-plugin/dolphinscheduler-storage-api/src/main/java/org/apache/dolphinscheduler/plugin/storage/api/StorageOperator.java]: The `udfs` have been remove from `ResourceType`.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/resources/visitor/ResourceTreeVisitor.java (hunk 2 above)
Comment (SbloodyS) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/resources/visitor/ResourceTreeVisitor.java]: Done.

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by caishunfeng — APPROVED** [review]
+1

**Pull request merged**